pennylane/ops/mid_measure/pauli_measure.py

```diff
@@ -117,7 +117,54 @@ def _pauli_measure_primitive_abstract_eval(*_, **__):
 
 
 def pauli_measure(pauli_word: str, wires: WiresLike, postselect: int | None = None):
-    """Perform a Pauli product measurement."""
+    """Perform a Pauli product measurement.
+
+    .. note::
+
+        The circuits generated from this function are currently not executable on any backend.
+        This function is only for analysis and potential future execution when a suitable backend is
+        available.
+
+    Args:
+        pauli_word (str): The Pauli word to measure.
+        wires (Wires): The wires the Pauli operators act on.
+        postselect (Optional[int]): The postselection value, one of 0 or 1. It projects the state into a
+            subspace of degenerated states whose eigenvalues are 1 or -1, respectively. None by default.
+
+    Returns:
+        MeasurementValue: A reference to the future result of the Pauli product measurement
+
+    Raises:
+        ValueError: if the Pauli word has characters other than X, Y and Z.
+
+    The following example illustrates how to include a Pauli product measurement (PPM) in a circuit by specifiying
+    the Pauli word and the wires it acts on:
+
+    **Example:**
+
+    .. code-block:: python
+
+        @qml.qnode(qml.device("null.qubit", wires=3))
+        def circuit():
+            qml.Hadamard(0)
+            qml.Hadamard(2)
+
+            ppm = qml.pauli_measure(pauli_word="XY", wires=[0, 2])
+            qml.cond(ppm, qml.X)(wires=1)
+
+            return qml.expval(qml.Z(0))
+
+    The PauliX operation on wire ``1`` will be applied conditionally on the value of the PPM outcome:
+
+    >>> print(qml.draw(circuit)())
+    0: ──H─╭┤↗X├────┤  <Z>
+    1: ────│──────X─┤
+    2: ──H─╰┤↗Y├──║─┤
+             ╚════╝
+
+    .. seealso::
+        :func:`catalyst.passes.ppm_compilation`
```

Comment:
This may be a better reference: https://pennylane.ai/compilation/pauli-product-measurement

Reply:
The func reference was suggested by Isaac, likely to link to the existing passes in Catalyst. But I can include both references with additional context